src/drivers/CensusCacheDriver.ts

```diff
@@ -7,29 +7,28 @@ import {Redis} from 'ioredis';
 
 @injectable()
 export default class CensusCacheDriver implements CacheContract {
-    private readonly namespace: string = 'census';
-    private readonly expiry: number = 86400; // 1 day
     private readonly cacheClient: Redis;
-    private readonly retrieving = new Map<string, Promise<any>>();
 
     constructor(
-        namespace: string,
-        expiry: number,
-        @inject(RedisConnection) cacheClient: RedisConnection,
+        @inject(RedisConnection) private readonly cacheConnection: RedisConnection,
```

Comment:
Same as `CensusStream`, maybe just inject a Redis instance from IoRedis? When the application start first it bootstraps, creating all the instances, and then it boots, initializing everything. Currently the connection to Redis is made during the bootstrap process and it makes the RedisConnectionService redundant. Besides you do not have to check if the connection is already initialized, because it happens in the RedisConnectionService and no where else.